### ISSUE TITLE
[Soundcloud] 64k opus higher quality than 128k mp3

### ISSUE THREAD

**Issue by Commandtechno** [yt-dlp/yt-dlp#6668]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Example URLs

can be any, heres just the one i used to test: https://soundcloud.com/hykeemcarter/killstreaks-feat-don-toliver

### Provide a description that is worded well enough to be understood

when downloading audio from soundcloud, yt-dlp prefers the 128k opus format to the 64k opus format that soundcloud provides for best audio quality

![available qualities](https://user-images.githubusercontent.com/68407783/228401110-99fa2450-0d58-4b2c-b86a-ea6a2925e486.png)

when using spek (https://www.spek.cc/) to analyze the spectrums, the 128k mp3 outputs a file peaking at 16khz (128kbps) while the 64k opus outputs a file peaking at 20khz (320kbps)

![mp3 spek analysis](https://user-images.githubusercontent.com/68407783/228401301-ad1c47d1-50df-43bd-be6b-43833ef827fd.png)

![opus spek analysis](https://user-images.githubusercontent.com/68407783/228401328-5d5c63d9-66df-4b82-b60c-83f4cb7778de.png)

should yt-dlp prefer opus as it outputs a higher quality file despite being a lower bitrate?

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
C:\Users\Techno\Downloads\fs-new>yt-dlp -FvU "https://soundcloud.com/hykeemcarter/killstreaks-feat-don-toliver"
[debug] Command-line config: ['-FvU', 'https://soundcloud.com/hykeemcarter/killstreaks-feat-don-toliver']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 5.1-essentials_build-www.gyan.dev (setts), ffprobe 5.1-essentials_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 5590c57bd0433ed239a2deaaf92e2ad6f37fe50f53664c821575cafe106a9421
yt-dlp is up to date (stable@2023.03.04)
[soundcloud] Extracting URL: https://soundcloud.com/hykeemcarter/killstreaks-feat-don-toliver
[soundcloud] hykeemcarter/killstreaks-feat-don-toliver: Downloading info JSON
[soundcloud] 1371553090: Downloading JSON metadata
[soundcloud] 1371553090: Downloading JSON metadata
[soundcloud] 1371553090: Downloading JSON metadata
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 1371553090:
ID           EXT  RESOLUTION │ PROTO │ VCODEC     ACODEC   ABR
──────────────────────────────────────────────────────────────
hls_opus_64  opus audio only │ m3u8  │ audio only unknown  64k
hls_mp3_128  mp3  audio only │ m3u8  │ audio only unknown 128k
http_mp3_128 mp3  audio only │ http  │ audio only unknown 128k

C:\Users\Techno\Downloads\fs-new>yt-dlp -vU "https://soundcloud.com/hykeemcarter/killstreaks-feat-don-toliver"
[debug] Command-line config: ['-vU', 'https://soundcloud.com/hykeemcarter/killstreaks-feat-don-toliver']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 5.1-essentials_build-www.gyan.dev (setts), ffprobe 5.1-essentials_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 5590c57bd0433ed239a2deaaf92e2ad6f37fe50f53664c821575cafe106a9421
yt-dlp is up to date (stable@2023.03.04)
[soundcloud] Extracting URL: https://soundcloud.com/hykeemcarter/killstreaks-feat-don-toliver
[soundcloud] hykeemcarter/killstreaks-feat-don-toliver: Downloading info JSON
[soundcloud] 1371553090: Downloading JSON metadata
[soundcloud] 1371553090: Downloading JSON metadata
[soundcloud] 1371553090: Downloading JSON metadata
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 1371553090: Downloading 1 format(s): http_mp3_128
[debug] Invoking http downloader on "https://cf-media.sndcdn.com/UUVklxNtpKAj.128.mp3?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLW1lZGlhLnNuZGNkbi5jb20vVVVWa2x4TnRwS0FqLjEyOC5tcDMqIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjgwMDUzMDk0fX19XX0_&Signature=cYqRz7Uzto2EWotGjfDtRJ0zLXlSiNKdOJictOvWO0jAlVv7U7r840EvplpQ6sVBXcTIDehHlUkaCdOZQG6WaBFwUrAMV2o4E8LutrIKeDnWKvnuFscwVGbVEfZYnbykgQU4HlQri54MS0qcnFptxhfNaoOaXOuCFBllmS2NM9dJaFPdhBXSwE6Aw0~-vwoYEAEHIcaFG43crVTSRWWFQCbD8vpsC6EjumXRBBTc-~8qeQcqMyX18Ue1ENG8DPso1riElvuVYfkG0F9EWVcNOshtoiw8QUeEgurqHPJL4D6SXfjOZTSLCYuLjs1or0EBLc6EW6YE4RFUyYkErjHuFw__&Key-Pair-Id=APKAI6TU7MMXM5DG6EPQ"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: killstreaks (feat. Don Toliver & PinkPantheress) [1371553090].mp3
[download] 100% of    3.29MiB in 00:00:00 at 4.11MiB/s

C:\Users\Techno\Downloads\fs-new>
```

**Comment by pukkandan** [comment]
Issue is that `acodec` is not being extracted. If extractor implements it, sorting will fix itself. Alternately, it may be reasonable to have some core code to guess the codec from ext

**Comment by bashonly** [comment]
Closed by 86a972033e05fea80e5fe7f2aff6723dbe2f3952 and a2d0840739cddd585d24e0ce4796394fc8a4fa2e
